core/server/master/src/main/java/alluxio/master/file/meta/MutableInode.java

```diff
@@ -70,6 +73,7 @@ protected MutableInode(long id, boolean isDirectory) {
     mPinned = false;
     mAcl = new AccessControlList();
     mUfsFingerprint = Constants.INVALID_UFS_FINGERPRINT;
+    mXAttr = new HashMap<>(1);
```

Comment:
we may want to consider not creating the object if it is unused, since java map objects take up some mem.

Reply:
I actually think that is a good idea, however I'm concerned about users getting back a null value when calling `getXAttr()`. If it is null, could we return an empty map there? or do you think we should just return `null`?